### ISSUE TITLE
Protocol checks failed

### ISSUE THREAD

**Issue by rzwq** [astral-sh/ty#3545]
### Summary

- Command (also using **ty** plugin in VS Code): `ty check module.py`

- No `pyproject.toml` used.

---

<img width="2386" height="722" alt="Image" src="https://github.com/user-attachments/assets/4f96d13f-7c85-43ee-b749-268df61c70e2" />

<img width="2356" height="990" alt="Image" src="https://github.com/user-attachments/assets/5c4b22b4-1df8-4192-813b-af75edab159f" />

```python
from typing import Protocol


class Comparable(Protocol):
    def __eq__(self, other) -> bool: ...
    def __ne__(self, other) -> bool: ...
    def __le__(self, other) -> bool: ...
    def __lt__(self, other) -> bool: ...
    def __ge__(self, other) -> bool: ...
    def __gt__(self, other) -> bool: ...


def lt(x: Comparable, y: Comparable) -> bool:
    return x < y


if __name__ == "__main__":
    # failed to check compare methods (false negative, like pyright)
    # mypy check successful
    print(lt(3, 5))  
```

### Version

0.0.39

**Comment by carljm** [comment]
Hi, thanks for the report!

ty is correct here (and so is pyright); the Python `int` type, as defined in typeshed, is not a subtype of your `Comparable` protocol. The reason is right there in ty's error message: the [comparison methods](https://github.com/python/typeshed/blob/main/stdlib/builtins.pyi#L351-L356) of `int` accept a positional-only parameter (cannot be called with a keyword argument), whereas your `Comparable` protocol requires those methods to accept a positional-or-keyword parameter named `other`.

mypy does not correctly enforce assignability of callables based on positional-only vs keyword-or-positional arguments. I'm not sure to what extent this is a considered decision, vs to what extent it is a legacy behavior because mypy existed long before positional-only parameters existed.

Modify your `Comparable` protocol to use positional-only parameters, and your code passes fine in ty and pyright:

```py
class Comparable(Protocol):
    def __eq__(self, other, /) -> bool: ...
    def __ne__(self, other, /) -> bool: ...
    def __le__(self, other, /) -> bool: ...
    def __lt__(self, other, /) -> bool: ...
    def __ge__(self, other, /) -> bool: ...
    def __gt__(self, other, /) -> bool: ...
```

**Comment by rzwq** [comment]
> The more I know **Python**, the less I know **Python**.

Thank you for the products you produce.
